activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb

```diff
@@ -539,7 +539,7 @@ def translate_exception(exception, message)
           end
         end
 
-        COLLATE_REGEX = /.*\"(\w+)\".*collate\s+\"(\w+)\".*/i.freeze
+        COLLATE_REGEX = /.*\"(\w+)\".*collate\s+\"(\w+)\".*/i
```

Comment:
`/.*\"(\w+)\".*collate\s+\"(\w+)\".*/i` is not a string so this `.freeze` is not redundant.

Reply:
Thanks for the review. Let me confirm this review first. I think you mean regular expressions is not a string. It looks like Ruby itself does not support frozen regular expressions yet.
https://bugs.ruby-lang.org/issues/8948 I'm wondering if this .freeze is actually valid or not

Reply:
While it does sound like Rubocop's being over-eager in declaring it redundant, I think this instance is fine in practice. It's not a value we return anywhere, so someone would have to be trying pretty hard to even find it... and at that point a `const_set` is just as easy.

Reply:
Yeah, this instance is fine in practice, and this is an only place where RuboCop's auto-correct affects in I reviewed all diff in this PR.
I suppose it is an issue for the cop, but I'm not sure whether RuboCop intentionally regard a regex literal like a string literal or not.

Reply:
I have opened a RuboCop issue https://github.com/rubocop-hq/rubocop/issues/6331

Reply:
Thank you for opening the RuboCop issue. I think this is a false positive of RuboCop. I opened a PR rubocop-hq/rubocop#6333 to fix, but it may take some time to the next new release of RuboCop (maybe 0.59.3 or 0.60.0) and subsequent Code Climate support. Until then it will depend on hand craft.